includes/Services/FlowService.php

```diff
@@ -0,0 +1,199 @@
+			if(strcmp(gettype($value), gettype($flow_data[$key])) != 0) {
+				$updated_flow_data[$key] = $value;
+				continue;
+			}
+
+			// Any Key renamed is updated in the database with NewKey and the value from the OldKey is retained or not based on retain_existing_value
+			if($flow_data_fixes) {
+				foreach ($flow_data_fixes as $old_key=>$replace_key) {
+					if (array_key_exists($old_key, $flow_data) && array_key_exists($replace_key['new_key'], $default_flow_data)) {
+						($replace_key['retain_existing_value'])?
+							$flow_data[$replace_key['new_key']] = $flow_data[$old_key]
+							: $flow_data[$replace_key['new_key']] = $default_flow_data[$replace_key['new_key']];
+						unset($flow_data[$old_key]);
+					}	
+				}
+			}
+
+			if(array_key_exists($key, $flow_data)) {
+
+				// Accepts the value of Exception List keys as is from the database OR non array OR empty array values
+				if(($exception_list && isset($exception_list[$key])) || !is_array($value) || (is_array($value) && sizeof($value) === 0))
+					$updated_flow_data[$key] = $flow_data[$key];
+
+				else {
+					// To handle Indexed Arrays gracefully
+					if (count(array_filter(array_keys($value), 'is_string')) === 0) {
+						// To check if an Indexed Array is further Nested or Not
+						foreach($value as $index_key => $index_value) {
+							// For Indexed Arrays having Non Associative Array Values
+							(!is_array($index_value) && isset($flow_data[$key]))? 
+								$updated_flow_data[$key] = $flow_data[$key] 
+								: $updated_flow_data[$key] = $value;
+						}
+						continue;
+					} 
+
+					// To handle Associative Arrays gracefully
+					$updated_flow_data[$key] = self::update_flow_data_recursive($value, $flow_data[$key]);
+				}
+			}
+
+			// Adds or deletes key-value pairs in DB based on the modification in the blueprint
+			else
+				$updated_flow_data[$key] = $value;
+		}
+		return $updated_flow_data;
+	}
+
+	/*
+	 * function to update the Database recursively based on Values opted or entered by the User
+	 */	
+	private static function update_post_call_data_recursive(&$flow_data, $params) {
+		static $mismatch_data_type = '';
+		$exception_list = Flows::get_exception_list();
+
+		foreach ($flow_data as $key => $value)
+		{ 		
+			// Updates value entered by the user
+			if (array_key_exists($key, $params)){			
+
+				// Accepts the value of Exception List keys as is from the database
+				if($exception_list && isset($exception_list[$key])) {
+					$flow_data[$key] = $params[$key];
+					continue;
+				}
+
+				// Error thrown if the datatype of the parameter does not match
+				if(strcmp(gettype($value), gettype($params[$key])) != 0) {
+					$mismatch_data_type = $key. ' => '. gettype($params[$key]) . '. Expected: ' . gettype($value);
+					break;
+				}
+
+				// Accepts non-Array Values entered by the user
+				elseif(!is_array($value))
+					$flow_data[$key] = $params[$key];
```

Comment:
```suggestion
if (array_key_exists($key, $params)){			
				$flow_data[$key] = $value;
			}

			// Accepts the value of Exception List keys as is from the database
			if($exception_list && isset($exception_list[$key])) {
				$flow_data[$key] = $params[$key];
				continue;
			}

			// Error thrown if the datatype of the parameter does not match
			if(strcmp(gettype($value), gettype($params[$key])) != 0) {
				$mismatch_data_type = $key. ' => '. gettype($params[$key]) . '. Expected: ' . gettype($value);
				break;
			}

			// Accepts non-Array Values entered by the user
			if(!is_array($value)) {
				$flow_data[$key] = $params[$key];
				continue;
			}

			// To handle Indexed Arrays gracefully
			if (count(array_filter(array_keys($params[$key]), 'is_string')) === 0 ) {
				// If the Database value is empty or an indexed Array, to avoid Associative arrays to be overwritten (Eg: data)
				((sizeof($value) === 0) || (count(array_filter(array_keys($value), 'is_string')) === 0))?
					$flow_data[$key] = $params[$key]
					: $flow_data[$key] = $value;
				continue;
			}

			// To handle Associative Arrays gracefully
			$flow_data[$key] = self::update_post_call_data_recursive($value, $params[$key]);
										
```
Again, lets make the code itself readable instead of adding comments on most lines